conan/internal/model/cpp_info.py

```diff
@@ -836,7 +836,7 @@ def required_components(self):
                 if r not in ret:
                     ret.append(r)
         # Then split the names
-        ret = [r.split("::") if "::" in r else (None, r) for r in ret]
+        ret = [r.split("::", 1) if "::" in r else (None, r) for r in ret]
```

Comment:
I would have put 
```py
r.split("::", maxsplit=1)
```
As that `1` may create confusion for the reader! 
But looks good to me!